components/brave_wallet/browser/eth_tx_manager_unittest.cc

```diff
@@ -1021,10 +1021,10 @@ TEST_F(EthTxManagerUnitTest, GetNonceForHardwareTransaction) {
           [&](const absl::optional<std::string>& result) {
             EXPECT_EQ(
                 result,
-                "0xf873808517fcf1832182960494be862ad9abfe6f22bcb087716c7d89a2"
-                "6051f74c88016345785d8a0000b844095ea7b30000000000000000000000"
-                "00bfb30a082f650c2a15d0632f0e87be4f8e64460f000000000000000000"
-                "0000000000000000000000000000003fffffffffffffff8205398080");
+                "0xf873018517fcf1832182960494be862ad9abfe6f22bcb087716c7d89a260"
+                "51f74c88016345785d8a0000b844095ea7b3000000000000000000000000bf"
+                "b30a082f650c2a15d0632f0e87be4f8e64460f000000000000000000000000"
+                "0000000000000000000000003fffffffffffffff8205398080");
```

Comment:
`npm run format`

Reply:
it's not just that, 0xf87380 vs 0xf87301

Reply:
There is an error in master https://github.com/brave/brave-core/blob/master/components/brave_wallet/browser/eth_nonce_tracker.cc#L94 it always returns true and loses error code in nonce tracker. Instead of expected 1 we got 0. Eth response parser returns parsing error because the payload was not hex https://github.com/brave/brave-core/pull/12539/files#diff-8d024761e2ddb9e2ff7eb58f36116810bf74adef4a80c4f1f894b2821bb04ff5R203.
In this PR I added error processing and changed the payload to be hex as expected by nonce.

Reply:
Got it, thanks.